subs/pantry/src/Pantry/Types.hs

```diff
@@ -148,6 +152,21 @@ data Package = Package
   }
   deriving (Show, Eq)
 
+data PackageCabal = PCCabalFile !TreeEntry -- ^ TreeEntry of Cabal file
+                  | PCHpack !TreeEntry -- ^ TreeEntry of HPack file
+                  | PCHpackCabalFile !TreeEntry -- ^ TreeEntry of Cabal file produced via HPack
+                    deriving (Show, Eq)
+
+getPCBlobKey :: PackageCabal -> BlobKey
```

Comment:
Similarly, this makes me very nervous: it's a function which silently gives either a cabal file or an hpack file. We're throwing away a lot of type safety.

Reply:
This function and it's usage has been removed altogether.